cp1404practicals/prac_09/sort_files_v2.py

```diff
@@ -0,0 +1,59 @@
+"""
+CP1404 Practical 9
+Sort files into directory depending on extension V2
+"""
+
+import shutil
+import os
+from collections import defaultdict
+
+
+def main():
+    """Demo os module functions."""
+    print("Starting directory is: {}".format(os.getcwd()))
+    # Change to desired directory
+    os.chdir('FilesToSort')
+    # Print a list of all files in current directory
+    print("Files in {}:\n{}\n".format(os.getcwd(), os.listdir('.')))
+
+    # create list of all file extensions
+    ext_list = []
+    filenames = os.listdir('.')
+    for filename in filenames:
+        ext = filename[filename.find('.'):].strip(".")
+        if ext in ext_list:
+            pass
+        else:
+            ext_list.append(ext)
+    print(ext_list)
+
+    # create list of category keys to associate with extensions based on user preference
+    # test functionality without needing user input: key_list = ['Docs', 'Docs', 'Images', 'Images', 'Docs', 'Spreadsheets', 'Spreadsheets', 'Images']
+    key_list = []
+    for ext in ext_list:
+        new_key = input("What category would you like to sort {} files into?".format(ext))
+        key_list.append(new_key)
```

Comment:
Possibly change this to
`ext_dict[ext] = new_key`
since this would create the dictionary at the same time